runtime/parachains/src/paras_inherent.rs

```diff
@@ -200,6 +200,11 @@ pub mod pallet {
 						.collect()
 				} else {
 					Vec::new()
+				};
+
+				if !freed_disputed.is_empty() {
+					freed_disputed.sort_unstable_by_key(|pair| pair.0); // sort by core index
```

Comment:
if the assumption of core uniqueness can be invalidated in the future (with "hyperthreads"?), I'd suggest using stable sort. Otherwise including the reasoning as a comment why using unstable is fine.

Reply:
It's used elsewhere and aligns with the implementers' guide. If we introduce hypertreads / fractional execution we will definitely need to revisit almost everything in the runtime in-depth.